kubernetes_cluster/01_main.tf

```
@@ -0,0 +1,194 @@
+  dns_prefix          = var.dns_prefix
+  kubernetes_version  = var.kubernetes_version
+  sku_tier            = var.sku_tier
+
+  private_cluster_enabled = var.private_cluster_enabled
+
+  #
+  # System node pool
+  #
+  default_node_pool {
+    name = var.system_node_pool_name
+
+    ### vm configuration
+    vm_size = var.system_node_pool_vm_size
+    # https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-general
+    os_disk_type                 = var.system_node_pool_os_disk_type # Managed or Ephemeral
+    os_disk_size_gb              = var.system_node_pool_os_disk_size_gb
+    type                         = "VirtualMachineScaleSets"
+    only_critical_addons_enabled = var.system_node_pool_only_critical_addons_enabled
+    availability_zones           = ["1", "2", "3"]
+    ultra_ssd_enabled            = var.system_node_pool_ultra_ssd_enabled
+    enable_host_encryption       = var.system_node_pool_enable_host_encryption
+
+    ### autoscaling
+    enable_auto_scaling = true
+    node_count          = var.system_node_pool_node_count_min
+    min_count           = var.system_node_pool_node_count_min
+    max_count           = var.system_node_pool_node_count_max
+
+    ### K8s node configuration
+    max_pods    = var.system_node_pool_max_pods
+    node_labels = var.system_node_pool_node_labels
+
+    ### networking
+    vnet_subnet_id        = var.vnet_subnet_id
+    enable_node_public_ip = false
+
+    upgrade_settings {
+      max_surge = var.upgrade_settings_max_surge
+    }
+
+    tags = merge(var.tags, var.system_node_pool_tags)
+  }
+
+  automatic_channel_upgrade       = var.automatic_channel_upgrade
+  api_server_authorized_ip_ranges = var.api_server_authorized_ip_ranges #tfsec:ignore:AZU008
+
+  # managed identity type: https://docs.microsoft.com/en-us/azure/aks/use-managed-identity
+  identity {
+    type = "SystemAssigned"
+  }
+
+  dynamic "network_profile" {
+    for_each = var.network_profile != null ? [var.network_profile] : []
+    iterator = p
+    content {
+      docker_bridge_cidr = p.value.docker_bridge_cidr
+      dns_service_ip     = p.value.dns_service_ip
+      network_policy     = p.value.network_policy
+      network_plugin     = p.value.network_plugin
+      outbound_type      = p.value.outbound_type
+      service_cidr       = p.value.service_cidr
+      load_balancer_sku  = "Standard"
+      load_balancer_profile {
+        outbound_ip_address_ids = var.outbound_ip_address_ids
+      }
+    }
+  }
+
+  role_based_access_control {
+    enabled = var.rbac_enabled
+    azure_active_directory {
+      managed                = true
+      admin_group_object_ids = var.aad_admin_group_ids
+    }
+  }
+
+  #
+  # Addons
+  #
+  addon_profile {
+    oms_agent {
+      enabled                    = var.log_analytics_workspace_id != null ? true : false #tfsec:ignore:AZU009
+      log_analytics_workspace_id = var.log_analytics_workspace_id
+    }
+    azure_keyvault_secrets_provider {
```

```suggestion
    key_vault_secrets_provider {
```
typo https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#key_vault_secrets_provider